src/Core/Services/AirtelMoneyService.cs

```diff
@@ -0,0 +1,85 @@
+            var referenceRegex = new Regex(@"(?<=Ref: )(.*?)(?=\s)");
+            var bankNameRegex = new Regex("(?<=from )(.*?)(?= on)");
+            var fromAgentRegex = new Regex("(Cash In)");
+            var amount = Decimal.Parse(amountRegex.Match(_message).ToString().Trim());
+            var reference = referenceRegex.Match(_message).ToString();
+            var fromAgent = fromAgentRegex.IsMatch(_message);
+            var BankName = bankNameRegex.Match(_message).ToString();
+
+            return new Payment()
+            {
+                Amount = amount,
+                PhoneNumber = _phoneNumber,
+                FromAgent = fromAgent,
+                Reference = reference,
+                BankName = IPaymentService.GetBankNameFromString(BankName),
+                ProviderName = Provider.Mpamba
+            };
+        }
+
+        public bool HasInvalidReference()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public bool IsDeposit()
+        {
+                        // sample messages for the regex below
+                        // var message = @" Trans ID: CI200808.0942.H11201: you have received MK20000.00 from BT59687, CHIMWEMWE MUGHOGHO. your new balance is MK20020.41.Note that Cash In is free of charge";
+                        // var message2 = "trans.ID :  PP200705.1009.H42097. Dear customer, you have received MK 6000.00 from 999025907,PRINCE CHIKWEBA . Your available balance is MK 6020.41.Trans id";
+                        // var message3 = "Dear Customer, money transfer to 999057130, ELIZABETH JERE is successful, Trans Id: PP200815.0822.G60250,Trans Amt: MK2000.00. Your available balance MK110.41";
+
+                        //other regex added for other functions 
+                        
+                        var findTrans = new Regex(@"([tT]rans ID: |[Tt]rans.ID|Trans Id:|Dear Customer)");
+                        var transIdRegex = new Regex(@"(?<=([tT]rans.ID :)|([tT]rans ID:|Trans Id:))(.*?)(?=([dD]|[yY]|:|,))");
+                        var amountRegex = new Regex(@"((?<=(Amt: MK |received MK|Trans Amt: MK))(.*?)(?=\.|[yY]|f))");
+                        var senderRegex = new Regex(@"(?<=(from |to ))(.*?)(?=\.|is)");
+                       
+
+
+                        var trans = findTrans.Match(_message).ToString();
+                        var transId = transIdRegex.Match(_message).ToString().Trim();
+                        var amount = amountRegex.Match(_message).ToString().Trim();
+                        var sender = senderRegex.Match(_message).ToString().Trim();
+                        var numberName = sender.Split(",");
+
+                        var number = numberName[0];
+                        var name = numberName[1].Trim();
+                        
+                        
+                        
+
+                        Console.WriteLine(trans + " " + transId + " " + amount + " " + sender + " " + number + " " + name);
```

Comment:
We do not do that here. All logs go through the logger https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1